TailLight/device.h

```diff
@@ -3,11 +3,27 @@
 /** Driver-specific struct for storing instance-specific data. */
 struct DEVICE_CONTEXT {
     UNICODE_STRING PdoName;
+    BOOLEAN        fSetBlackSuccess;
     WDFWMIINSTANCE WmiInstance;
     UNICODE_STRING HardwareId;
 };
+
+
+template<typename T> inline void NukeWdfHandle(T& handle) {
```

Comment:
This tiny function is only called from two places. Therefore suggest that you remove it and instead call `WdfObjectDelete(...)` and so on directly from the client code.